clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/ops/UpsertAzureLoadBalancerAtomicOperation.groovy

```diff
@@ -51,25 +50,40 @@ class UpsertAzureLoadBalancerAtomicOperation implements AtomicOperation<Map> {
     task.updateStatus BASE_PHASE, "Initializing upsert of load balancer $description.loadBalancerName " +
       "in $description.region..."
 
+    Map<String, Boolean> resourceCompletedState = new HashMap<String, Boolean>()
+
     try {
-      DeploymentExtended deployment = description.credentials.resourceManagerClient.createLoadBalancerFromTemplate(description.credentials,
+
+      task.updateStatus(BASE_PHASE, "Beginning load balancer deployment")
+
+      String resourceGroupName = AzureUtilities.getResourceGroupName(description.appName, description.region)
```

Comment:
The generation of the resource group name should be done in CreateResourceFromTemplate method in AzureResourceManagerClient, otherwise you are going to have to duplicate this code before every call to this method.